src/lib/ProjectBasics.svelte

```diff
@@ -33,66 +39,85 @@ const countryList = Object.values(countries);
-    'Art & Culture', 'Community Development', 'Innovation'
-  ];
-
-  let navSections = [
-    { id: 'basics', label: 'Basics', href: '/createProject' },
-    { id: 'team', label: 'Team', href: '/createProject/team' },
-    { id: 'links', label: 'Links', href: '/createProject/links' },
-    { id: 'funding', label: 'Funding', href: '/createProject/funding' }
-  ];
+  let availableTags = [];
 
   let currentSection = 'basics';
 
-  function handleNavigation(event) {
-  const { href, id } = event.detail;
-  currentSection = id;
-  goto(href);
-}
-
   function toggleDropdown(event) {
     event.preventDefault();
     isOpen = !isOpen;
   }
 
   function addTag(tag) {
-    if (!selectedTags.includes(tag)) {
+    if (!selectedTags.some(selected => selected.title === tag.title)) {
       selectedTags = [...selectedTags, tag];
+      updateStore();
       dispatch('change', selectedTags);
     }
+
     inputValue = '';
     isOpen = false;
   }
 
   function removeTag(tag) {
-    selectedTags = selectedTags.filter(t => t !== tag);
+    selectedTags = selectedTags.filter(t => t.title !== tag.title);
+    updateStore();
     dispatch('change', selectedTags);
+    console.log(selectedTags);
   }
 
   $: filteredTags = availableTags.filter(tag => 
-    tag.toLowerCase().includes(inputValue.toLowerCase()) && !selectedTags.includes(tag)
+    tag.title.toLowerCase().includes(inputValue.toLowerCase()) && 
+    !selectedTags.some(selected => selected.title === tag.title)
   );
 
 
+
   let ProjectBannerImage = null;
-    let ProjectProfileImage = null;
+  let ProjectProfileImage = null;
+
+  const authorizedExtensions = ['.jpg', '.jpeg', '.png', '.webp'];
   
-    function handleBannerUpload(event) {
-      const file = event.target.files[0];
-      if (file) {
-        ProjectBannerImage = URL.createObjectURL(file);
-      }
+  function handleBannerUpload(event) {
+    const file = event.target.files[0];
+    if (file) {
+      ProjectBannerImage = URL.createObjectURL(file);
     }
-  
-    function handleProfileUpload(event) {
-      const file = event.target.files[0];
-      if (file) {
-        ProjectProfileImage = URL.createObjectURL(file);
+  }
+
+  function handleProfileUpload(event) {
+    const file = event.target.files[0];
+    if (file) {
+      ProjectProfileImage = URL.createObjectURL(file);
+    }
+  }
+
+
+  async function fetchAllCategories() {
+    try {
+      const response = await fetch('/api/categories', {
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json'
+        }
+      });
+
+      if (!response.ok) {
+        throw new Error(response.statusText);
       }
+
+      const data = await response.json();
+      availableTags = data.categories;
+    } catch (error) {
+      
     }
+  }
+
+    onMount(async () => {
+      fetchAllCategories();
+      const module = await import("novel-svelte");
+      Editor = module.Editor;
+   });
```

Comment:
alternative way to handle data fetching:
https://svelte.dev/tutorial/kit/page-data